### DISCUSSION TITLE
Running multiple instances of umtprd

### DISCUSSION THREAD

**Discussion by marcone** [viveris/uMTP-Responder#117] [General]
Technically it's possible to run multiple instances of umtprd on separate UDCs.
Right now this requires using separate umtprd binaries (because the message queue id is keyed to the canonical path of the umtprd binary being run), and I haven't actually come across any devices that have multiple working UDCs (I did come across a device that had two entries in /sys/class/udc/, but only one actually worked)

@jfdelnero is supporting multiple UDCs something you actually care about? I've been playing around with replacing the SYSV message queue with a POSIX message queue, but it would need some additional work if it needs to support multiple concurrent umtprd instances.

**Comment by jfdelnero** [comment]
Have you a device with multiple UDC ?

**Reply by marcone** [reply]
> Have you a device with multiple UDC ?

No, like I said, I haven't actually come across any devices that have multiple working UDCs, though I did come across a device that had multiple entries in /sys/class/udc, but only one of those actually worked (I think it was a bad device tree in that particular linux image).
